components/prompt_based_laion_retrieval/src/main.py

```diff
@@ -78,17 +78,6 @@ def transform(
 
         # unpack list of urls
         dataframe = dataframe.explode("images_url")
-
-        # add id and source columns
-        # TODO use LAION id instead
-        dataframe["id"] = dataframe.assign(id=1).id.cumsum()
-        dataframe["source"] = "laion"
-
-        # reorder columns
-        dataframe = dataframe[["id", "source", "images_url"]]
-
-        dataframe = dataframe.astype({'id': 'string', 'source': 'string'})
-
         dataframe = dataframe.reset_index(drop=True)
```

Comment:
This creates the default index, could we not do it dynamically when adding rows ?